### PR TITLE
Document priority queue, max heap. Fixes: #2901

### PR DESCRIPTION
**Pull request by LeStarch**
| | |
|:---|:---|
|**_Related Issue(s)_**|  |
|**_Has Unit Tests (y/n)_**|  |
|**_Documentation Included (y/n)_**|  |

---
## Change Description

Documents algorithms in Os::Generic, fixing #2901

### CODE REVIEW

**Comment by LeStarch** [comment]
Spell check is failing on a miss-spelling that isn't there.  Not sure how to fix.

File: Os/Generic/docs/sdd.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+In order to prioritize messages, a [Types::MaxHeap](#typesmaxheap-data-structure) data structure is used.
+
+When a message is received from a calling sender, `find_index` returns a free index from the free list. The data is copied into the message data array, and the size into the size array using this free index via `store_data`. The index is then inserted into the max heap structure for prioritization. When the queue is full and the `BLOCKING` option was supplied, the sender will block on the `m_full` condition variable until notified of a dequeue.
+
+When a message is dequeued, the highest priority index is removed from the max heap. The data is copied out from the data array, and the size from the size array using that index via `load_data` and sent to the calling receiver. The index is then returned to the free list via `return_index` to indicate that it may be reused. When the queue is empty and the `BLOCKING` option was supplied, the receiver will block on the `m_empty` condition variable until notified of a dequeue.
```
Comment (thomas-bc) [inline Os/Generic/docs/sdd.md]: > When the queue is empty and the `BLOCKING` option was supplied, the receiver will block on the `m_empty` condition variable until notified of a **dequeue**.

Should this read "_until notified of an **enqueue**_" ?

**Review by thomas-bc — CHANGES_REQUESTED** [review]
Really nice! Just one comment

File: Os/Generic/docs/sdd.md (hunk 1 above)
Comment (LeStarch) [reply Os/Generic/docs/sdd.md]: Fixed!

**Review by thomas-bc — APPROVED** [review]

**Pull request merged**